### PR TITLE
add PullVariable/PullEmbeddingVector rpc in go

### LINKED ISSUE #1654: Implement PullEmbeddingVector grpc service in Go

### LINKED ISSUE #1655: Implement PullVariable grpc service in Go

### PR DESCRIPTION
**Pull request by QiJune**
Fix #1654 
Fix #1655

### CODE REVIEW

File: elasticdl/pkg/common/embedding_table.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ func (table *EmbeddingTable) GetEmbeddingVector(index int64) *Tensor {
 	}
 	// TODO(qijun) only support zero initializer now
 	newVector := NewVector(table.Dim)
+	newVector.Name = table.Name
```
Comment (wangkuiyi) [inline elasticdl/pkg/common/embedding_table.go]: How about `newVector := NewVecotr(table.Dim, table.Name)`?

File: elasticdl/pkg/ps/server.go [hunk 2]
```diff
@@ -2,52 +2,110 @@ package ps
...
+	"sync"
 )
 
-type psServer struct {
+// Server defines servicer of ps
+type Server struct {
 	pb.PserverServer
+	Param *Parameter
+	Opt   Optimizer
+	PSID  int32
```
Comment (wangkuiyi) [inline elasticdl/pkg/ps/server.go]: PSID => ID, because the fullname is `ps.Server.ID`

File: elasticdl/pkg/ps/server.go [hunk 3]
```diff
@@ -2,52 +2,110 @@ package ps
...
+	lock  *sync.Mutex
 }
 
-func (s *psServer) PullVariable(ctx context.Context, in *pb.PullVariableRequest) (*pb.PullVariableResponse, error) {
-	// TODO: implement the service.
-	return &pb.PullVariableResponse{}, nil
+// NewServer creates a Server instance
+func NewServer(PSID int32, opt string, lr float32) *Server {
```
Comment (wangkuiyi) [inline elasticdl/pkg/ps/server.go]: Need a comment about ID (PSID) -- does it have to be a zero-based successive integer number, or any number, e.g., an MD5 hash value? If it is the latter case, users should not have to specify it in the NewServer function.

File: elasticdl/pkg/ps/server_test.go [hunk 4]
```diff
@@ -22,61 +27,171 @@ func createClient() (pb.PserverClient, context.Context, *grpc.ClientConn, contex
...
-		t.Errorf("Failed to pull variable")
+		t.Errorf("Failed to push model")
 	}
+
+	assert.True(t, s.Param.InitStatus)
+	assert.Len(t, s.Param.NonEmbeddingParam, 2)
+	assert.Contains(t, s.Param.NonEmbeddingParam, "t1")
+	assert.Contains(t, s.Param.NonEmbeddingParam, "t2")
+	assert.True(t, common.CompareFloatArray(a, s.Param.GetNonEmbeddingParam("t1").Value, 0.001))
+	assert.True(t, common.CompareFloatArray(b, s.Param.GetNonEmbeddingParam("t2").Value, 0.001))
```
Comment (mhaoli) [inline elasticdl/pkg/ps/server_test.go:63]: Do we need to check parameter version here?

File: elasticdl/pkg/ps/server.go [hunk 5]
```diff
@@ -2,52 +2,110 @@ package ps
...
+	if !s.Param.InitStatus {
+		res.ModelInitStatus = false
+		return &res, nil
+	}
+	res.ModelInitStatus = true
+	res.Model = &pb.Model{}
+	res.Model.Version = s.Param.Version
+	if s.Param.Version > in.CurrentModelVersion {
+		for _, v := range s.Param.NonEmbeddingParam {
+			fmt.Println(v.Value)
```
Comment (mhaoli) [inline elasticdl/pkg/ps/server.go]: Remove fmt.Println here?

Comment (QiJune) [reply elasticdl/pkg/ps/server.go]: Done

File: elasticdl/pkg/ps/server.go (hunk 3 above)
Comment (QiJune) [reply elasticdl/pkg/ps/server.go]: Done, it's the former case,  add comments.

File: elasticdl/pkg/ps/server.go (hunk 2 above)
Comment (QiJune) [reply elasticdl/pkg/ps/server.go]: Done

File: elasticdl/pkg/common/embedding_table.go (hunk 1 above)
Comment (QiJune) [reply elasticdl/pkg/common/embedding_table.go]: Done

File: elasticdl/pkg/ps/server_test.go (hunk 4 above)
Comment (QiJune) [reply elasticdl/pkg/ps/server_test.go:63]: The model version in PS is directly set to the version in request. 

The key point of PushModel rpc is actually `InitStatus`. Several workers may push model to one PS, and once the model from one worker is  accepted, the others will be skipped.

We  have check the `InitStatus` already in this unit test.

**Review by mhaoli — APPROVED** [review]

**Pull request merged**